From decb3c61c23ffcd98828a7b82c14b9a1981c6288 Mon Sep 17 00:00:00 2001
From: Valentine Barshak <valentine.barshak@cogentembedded.com>
Date: Fri, 26 Oct 2018 13:41:47 +0300
Subject: [PATCH 201/248] arm64: dts: renesas: r8a77970: Remove LVDS encoder
 nodes

This removes LVDS encoder nodes since we don't support separate
LVDS encoder driver which has been recently introduced in the
Renesas Kernel Development Tree (kernel/git/horms/renesas.git).

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
---
 arch/arm64/boot/dts/renesas/r8a77970-eagle.dts |  4 ---
 arch/arm64/boot/dts/renesas/r8a77970-v3msk.dts | 23 +++++++--------
 arch/arm64/boot/dts/renesas/r8a77970.dtsi      | 41 ++++++--------------------
 3 files changed, 19 insertions(+), 49 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a77970-eagle.dts b/arch/arm64/boot/dts/renesas/r8a77970-eagle.dts
index 21f9cf5..6ba8b08 100644
--- a/arch/arm64/boot/dts/renesas/r8a77970-eagle.dts
+++ b/arch/arm64/boot/dts/renesas/r8a77970-eagle.dts
@@ -196,10 +196,6 @@
 
 &du {
 	status = "okay";
-};
-
-&lvds0 {
-	status = "okay";
 
 	ports {
 		port@1 {
diff --git a/arch/arm64/boot/dts/renesas/r8a77970-v3msk.dts b/arch/arm64/boot/dts/renesas/r8a77970-v3msk.dts
index bf2c7ab..8ada07c 100644
--- a/arch/arm64/boot/dts/renesas/r8a77970-v3msk.dts
+++ b/arch/arm64/boot/dts/renesas/r8a77970-v3msk.dts
@@ -116,9 +116,18 @@
 
 &du {
 	clocks = <&cpg CPG_MOD 724>,
+		 <&cpg CPG_MOD 727>,
 		 <&osc5_clk>;
-	clock-names = "du.0", "dclkin.0";
+	clock-names = "du.0", "lvds.0", "dclkin.0";
 	status = "okay";
+
+	ports {
+		port@1 {
+			lvds0_out: endpoint {
+				remote-endpoint = <&thc63lvd1024_in>;
+			};
+		};
+	};
 };
 
 &extal_clk {
@@ -198,18 +207,6 @@
 	};
 };
 
-&lvds0 {
-	status = "okay";
-
-	ports {
-		port@1 {
-			lvds0_out: endpoint {
-				remote-endpoint = <&thc63lvd1024_in>;
-			};
-		};
-	};
-};
-
 &mmc0 {
 	pinctrl-0 = <&mmc_pins>;
 	pinctrl-names = "default";
diff --git a/arch/arm64/boot/dts/renesas/r8a77970.dtsi b/arch/arm64/boot/dts/renesas/r8a77970.dtsi
index f609156..ae28386 100644
--- a/arch/arm64/boot/dts/renesas/r8a77970.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a77970.dtsi
@@ -1117,12 +1117,17 @@
 
 		du: display@feb00000 {
 			compatible = "renesas,du-r8a77970";
-			reg = <0 0xfeb00000 0 0x80000>;
+			reg = <0 0xfeb00000 0 0x80000>,
+			      <0 0xfeb90000 0 0x14>;
+			reg-names = "du.0", "lvds.0";
 			interrupts = <GIC_SPI 256 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 724>;
-			clock-names = "du.0";
+			clocks = <&cpg CPG_MOD 724>,
+				 <&cpg CPG_MOD 727>;
+			clock-names = "du.0", "lvds.0";
 			power-domains = <&sysc R8A77970_PD_ALWAYS_ON>;
-			resets = <&cpg 724>;
+			resets = <&cpg 724>,
+				 <&cpg 727>;
+			reset-names = "du.0", "lvds.0";
 			vsps = <&vspd0>;
 			status = "disabled";
 
@@ -1139,34 +1144,6 @@
 				port@1 {
 					reg = <1>;
 					du_out_lvds0: endpoint {
-						remote-endpoint = <&lvds0_in>;
-					};
-				};
-			};
-		};
-
-		lvds0: lvds-encoder@feb90000 {
-			compatible = "renesas,r8a77970-lvds";
-			reg = <0 0xfeb90000 0 0x14>;
-			clocks = <&cpg CPG_MOD 727>;
-			power-domains = <&sysc R8A77970_PD_ALWAYS_ON>;
-			resets = <&cpg 727>;
-			status = "disabled";
-
-			ports {
-				#address-cells = <1>;
-				#size-cells = <0>;
-
-				port@0 {
-					reg = <0>;
-					lvds0_in: endpoint {
-						remote-endpoint =
-							<&du_out_lvds0>;
-					};
-				};
-				port@1 {
-					reg = <1>;
-					lvds0_out: endpoint {
 					};
 				};
 			};
-- 
2.7.4

